client/connect-account-page/index.tsx

```diff
@@ -74,6 +117,125 @@ const ConnectAccountPage: React.FC = () => {
 		return source;
 	};
 
+	const updateLoaderProgress = ( maxPercent: number, step: number ) => {
+		if ( loaderProgressRef.current < maxPercent ) {
+			const newProgress = loaderProgressRef.current + step;
+			setTestDriveLoaderProgress( newProgress );
+		}
+	};
+
+	const checkAccountStatus = () => {
+		// Fetch account status from the cache.
+		apiFetch( {
+			path: `/wc/v3/payments/accounts`,
+			method: 'GET',
+		} ).then( ( account ) => {
+			// Fake update loader progress bar percentage.
+			updateLoaderProgress( 100, 5 );
+
+			// If the account status is complete, redirect to the overview page.
+			// Otherwise, schedule another check after 5 seconds.
+			if ( ( account as AccountData ).status === 'complete' ) {
```

Comment:
We should possibly look to have some sort of fallback logic in case the status doesn't reach complete. At the moment it looks like we'll loop around until it's complete, but I guess if there's a bug on Stripe's side and this doesn't happen, how should we handle it?

Reply:
That's a great idea. Thanks!